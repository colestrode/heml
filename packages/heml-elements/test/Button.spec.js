const render = require('@heml/render')
const Button = require('../build/button').default
const Style = require('../build/Style').default

beforeAll(() => {
  Style.preRender({ elements: [] });
})

test('renders with some attrs', () => {
  const attrs = {
    margin: '1px 2px 3px 4px'
  }

  return expect(render.renderElement(Button, attrs, 'Lady Dynamite')).toMatchSnapshot()
});
