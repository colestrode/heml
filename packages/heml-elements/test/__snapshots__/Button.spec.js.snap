// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with some attrs 1`] = `"<div margin=\\"1px 2px 3px 4px\\" class=\\"button\\"><table role=\\"presentation\\" width=\\"100%\\" align=\\"left\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td><table role=\\"presentation\\" width=\\"auto\\" align=\\"center\\" border=\\"0\\" cellspacing=\\"0\\" cellpadding=\\"0\\" class=\\"button__table\\"><tr><td align=\\"center\\" class=\\"button__cell\\"><a href=\\"#\\" class=\\"button__link\\" style=\\"display: inline-block;\\"><span class=\\"button__text\\">Lady Dynamite</span></a></td></tr></table></td></tr></table></div>"`;
